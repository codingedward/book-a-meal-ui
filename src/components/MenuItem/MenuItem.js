import React from 'react';
import './styles.css';


class MenuItem extends React.Component {

    render() {
        return (
            <div className="col-10 offset-1 offset-sm-0 col-sm-6 col-md-4 col-lg-3 text-center">
                <div className="menu-item pt-3 pl-3 pr-3 m-1">
                    <img src={this.props.item.img_url} className="img-fluid" alt="Food"/>
                    <table className="table">
                        <tbody>
                            <tr>
                                <td className="p-2">Ksh. {this.props.item.cost}</td>
                            </tr>
                            <tr>
                                <td className="p-2">{this.props.item.name}</td>
                            </tr>
                            <tr>
                                <td className="pb-0">
                                    <a href="login.html" className="btn btn-primary">Order Now</a>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
        );
    }
}

export default MenuItem;
