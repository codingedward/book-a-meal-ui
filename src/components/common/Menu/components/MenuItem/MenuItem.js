import React from 'react';
import { Link } from 'react-router-dom';
import './styles.css';

const MenuItem = ({ item }) => (
    <div className="col-10 offset-1 offset-sm-0 col-sm-6 col-md-4 col-lg-3 text-center">
        <div className="menu-item pt-3 pl-3 pr-3 m-1">
            <img src={item.img_url} className="img-fluid" alt="Food"/>
            <table className="table">
                <tbody>
                    <tr>
                        <td className="p-2">Ksh. {item.cost}</td>
                    </tr>
                    <tr>
                        <td className="p-2">{item.name}</td>
                    </tr>
                    <tr>
                        <td className="pb-0">
                            <Link to="/login" className="btn btn-primary"> Order Now</Link>
                        </td>
                    </tr>
                </tbody>
            </table>
        </div>
    </div>
);

export default MenuItem;
