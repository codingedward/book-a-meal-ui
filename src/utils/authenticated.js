export default function authenticated() {
    return localStorage.getItem('token') !== null;
}
