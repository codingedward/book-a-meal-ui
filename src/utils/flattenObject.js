export default function flattenObject(obj) {
    let result = {};
    for (const key in obj) {
        if (! obj.hasOwnProperty(key))
            continue;

        if ((typeof obj[key]) === 'object' && obj[key] !== null) {
            let flattened = flattenObject(obj[key])
            for (const key2 in flattened) { 
                if (! flattened.hasOwnProperty(key2))
                    continue;
                result[key + '.' + key2] = flattened[key2];
            }

        } else {
            result[key] = obj[key];
        }
    }

    return result;
}
