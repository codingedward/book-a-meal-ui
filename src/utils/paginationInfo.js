export default function paginationInfo(collection) {

    if (!collection || !collection.payload) {
        return {}
    }
    const { payload } = collection;
    let currentPage = 1;
    if (payload.next_page)
        currentPage = payload.next_page - 1;
    else if (payload.prev_page)
        currentPage = payload.prev_page + 1;

    return {
        hasNext: payload.has_next,
        hasPrev: payload.has_prev,
        nextPage: payload.next_page,
        prevPage: payload.prev_page,
        total: payload.total,
        pageCount: Math.ceil(payload.total / payload.per_page),
        perPage: payload.per_page,
        currentPage,
    };
}
