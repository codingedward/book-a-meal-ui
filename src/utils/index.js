import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';

export { singleError, paginationInfo, authenticated }

