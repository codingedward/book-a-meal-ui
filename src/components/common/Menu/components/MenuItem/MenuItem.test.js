import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; // order matters!
import MenuItem from './MenuItem';
import Adapter from 'enzyme-adapter-react-16';

const item = {
    name: 'name2',
    cost: 50,
    img_url: '#'
}

describe('<MenuItem />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<MenuItem item={item} />);
        expect(wrapper).toMatchSnapshot();
    });
});



