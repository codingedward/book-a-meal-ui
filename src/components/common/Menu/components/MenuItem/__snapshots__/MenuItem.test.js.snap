// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    item={
        Object {
            "cost": 50,
            "img_url": "#",
            "name": "name2",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="menu-item pt-3 pl-3 pr-3 m-1"
>
        <img
                alt="Food"
                className="img-fluid"
                src="#"
        />
        <table
                className="table"
        >
                <tbody>
                        <tr>
                                <td
                                        className="p-2"
                                >
                                        Ksh. 
                                        50
                                </td>
                        </tr>
                        <tr>
                                <td
                                        className="p-2"
                                >
                                        name2
                                </td>
                        </tr>
                        <tr>
                                <td
                                        className="pb-0"
                                >
                                        <Link
                                                className="btn btn-primary"
                                                replace={false}
                                                to="/login"
                                        >
                                                 Order Now
                                        </Link>
                                </td>
                        </tr>
                </tbody>
        </table>
</div>,
      "className": "col-10 offset-1 offset-sm-0 col-sm-6 col-md-4 col-lg-3 text-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Food"
            className="img-fluid"
            src="#"
/>,
          <table
            className="table"
>
            <tbody>
                        <tr>
                                    <td
                                                className="p-2"
                                    >
                                                Ksh. 
                                                50
                                    </td>
                        </tr>
                        <tr>
                                    <td
                                                className="p-2"
                                    >
                                                name2
                                    </td>
                        </tr>
                        <tr>
                                    <td
                                                className="pb-0"
                                    >
                                                <Link
                                                            className="btn btn-primary"
                                                            replace={false}
                                                            to="/login"
                                                >
                                                             Order Now
                                                </Link>
                                    </td>
                        </tr>
            </tbody>
</table>,
        ],
        "className": "menu-item pt-3 pl-3 pr-3 m-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Food",
            "className": "img-fluid",
            "src": "#",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tbody>
              <tr>
                            <td
                                          className="p-2"
                            >
                                          Ksh. 
                                          50
                            </td>
              </tr>
              <tr>
                            <td
                                          className="p-2"
                            >
                                          name2
                            </td>
              </tr>
              <tr>
                            <td
                                          className="pb-0"
                            >
                                          <Link
                                                        className="btn btn-primary"
                                                        replace={false}
                                                        to="/login"
                                          >
                                                         Order Now
                                          </Link>
                            </td>
              </tr>
</tbody>,
            "className": "table",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td
                                    className="p-2"
                  >
                                    Ksh. 
                                    50
                  </td>
</tr>,
                <tr>
                  <td
                                    className="p-2"
                  >
                                    name2
                  </td>
</tr>,
                <tr>
                  <td
                                    className="pb-0"
                  >
                                    <Link
                                                      className="btn btn-primary"
                                                      replace={false}
                                                      to="/login"
                                    >
                                                       Order Now
                                    </Link>
                  </td>
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td
                    className="p-2"
>
                    Ksh. 
                    50
</td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Ksh. ",
                      50,
                    ],
                    "className": "p-2",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Ksh. ",
                    50,
                  ],
                  "type": "td",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td
                    className="p-2"
>
                    name2
</td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "name2",
                    "className": "p-2",
                  },
                  "ref": null,
                  "rendered": "name2",
                  "type": "td",
                },
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <td
                    className="pb-0"
>
                    <Link
                                        className="btn btn-primary"
                                        replace={false}
                                        to="/login"
                    >
                                         Order Now
                    </Link>
</td>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="btn btn-primary"
                      replace={false}
                      to="/login"
>
                       Order Now
</Link>,
                    "className": "pb-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Order Now",
                      "className": "btn btn-primary",
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": " Order Now",
                    "type": [Function],
                  },
                  "type": "td",
                },
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
          "type": "table",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="menu-item pt-3 pl-3 pr-3 m-1"
>
          <img
                    alt="Food"
                    className="img-fluid"
                    src="#"
          />
          <table
                    className="table"
          >
                    <tbody>
                              <tr>
                                        <td
                                                  className="p-2"
                                        >
                                                  Ksh. 
                                                  50
                                        </td>
                              </tr>
                              <tr>
                                        <td
                                                  className="p-2"
                                        >
                                                  name2
                                        </td>
                              </tr>
                              <tr>
                                        <td
                                                  className="pb-0"
                                        >
                                                  <Link
                                                            className="btn btn-primary"
                                                            replace={false}
                                                            to="/login"
                                                  >
                                                             Order Now
                                                  </Link>
                                        </td>
                              </tr>
                    </tbody>
          </table>
</div>,
        "className": "col-10 offset-1 offset-sm-0 col-sm-6 col-md-4 col-lg-3 text-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Food"
              className="img-fluid"
              src="#"
/>,
            <table
              className="table"
>
              <tbody>
                            <tr>
                                          <td
                                                        className="p-2"
                                          >
                                                        Ksh. 
                                                        50
                                          </td>
                            </tr>
                            <tr>
                                          <td
                                                        className="p-2"
                                          >
                                                        name2
                                          </td>
                            </tr>
                            <tr>
                                          <td
                                                        className="pb-0"
                                          >
                                                        <Link
                                                                      className="btn btn-primary"
                                                                      replace={false}
                                                                      to="/login"
                                                        >
                                                                       Order Now
                                                        </Link>
                                          </td>
                            </tr>
              </tbody>
</table>,
          ],
          "className": "menu-item pt-3 pl-3 pr-3 m-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Food",
              "className": "img-fluid",
              "src": "#",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tbody>
                <tr>
                                <td
                                                className="p-2"
                                >
                                                Ksh. 
                                                50
                                </td>
                </tr>
                <tr>
                                <td
                                                className="p-2"
                                >
                                                name2
                                </td>
                </tr>
                <tr>
                                <td
                                                className="pb-0"
                                >
                                                <Link
                                                                className="btn btn-primary"
                                                                replace={false}
                                                                to="/login"
                                                >
                                                                 Order Now
                                                </Link>
                                </td>
                </tr>
</tbody>,
              "className": "table",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td
                                        className="p-2"
                    >
                                        Ksh. 
                                        50
                    </td>
</tr>,
                  <tr>
                    <td
                                        className="p-2"
                    >
                                        name2
                    </td>
</tr>,
                  <tr>
                    <td
                                        className="pb-0"
                    >
                                        <Link
                                                            className="btn btn-primary"
                                                            replace={false}
                                                            to="/login"
                                        >
                                                             Order Now
                                        </Link>
                    </td>
</tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <td
                      className="p-2"
>
                      Ksh. 
                      50
</td>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Ksh. ",
                        50,
                      ],
                      "className": "p-2",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Ksh. ",
                      50,
                    ],
                    "type": "td",
                  },
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <td
                      className="p-2"
>
                      name2
</td>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "name2",
                      "className": "p-2",
                    },
                    "ref": null,
                    "rendered": "name2",
                    "type": "td",
                  },
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <td
                      className="pb-0"
>
                      <Link
                                            className="btn btn-primary"
                                            replace={false}
                                            to="/login"
                      >
                                             Order Now
                      </Link>
</td>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="btn btn-primary"
                        replace={false}
                        to="/login"
>
                         Order Now
</Link>,
                      "className": "pb-0",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": " Order Now",
                        "className": "btn btn-primary",
                        "replace": false,
                        "to": "/login",
                      },
                      "ref": null,
                      "rendered": " Order Now",
                      "type": [Function],
                    },
                    "type": "td",
                  },
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
            "type": "table",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
